source/assets/js/playground/autocomplete.ts

```diff
@@ -0,0 +1,297 @@
+  'use',
+  'forward',
+  'import',
+  'mixin',
+  'include',
+  'function',
+  'extend',
+  'error',
+  'warn',
+  'debug',
+  'at-root',
+  'if',
+  'else',
+  'each',
+  'for',
+  'while',
+];
+
+const atRuleOptions = Object.freeze(
+  atRuleKeywords.map(keyword => ({
+    label: `@${keyword} `,
+    type: 'keyword',
+  }))
+);
+
+function atRuleCompletion(context: CompletionContext): CompletionResult | null {
+  const atRule = context.matchBefore(/@\w*/);
+  if (!atRule) return null;
+  if (atRule.from === atRule.to && !context.explicit) return null;
+  return {
+    from: atRule.from,
+    to: atRule.to,
+    options: atRuleOptions,
+  };
+}
+
+type CompletionInfo = {
+  name: string;
+  description?: string;
+};
+
+type ModuleDefinition = CompletionInfo & {
+  functions?: CompletionInfo[];
+  variables?: CompletionInfo[];
+};
+
+const builtinModules: ModuleDefinition[] = [
+  {
+    name: 'color',
+    description:
+      'generates new colors based on existing ones, making it easy to build color themes',
+  },
+  {
+    name: 'list',
+    description: 'lets you access and modify values in lists',
+    functions: [
+      {name: 'length'},
+      {name: 'nth'},
+      {name: 'set-nth'},
+      {name: 'join'},
+      {name: 'append'},
+      {name: 'zip'},
+      {name: 'index'},
+      {name: 'is-bracketed'},
+      {name: 'separator'},
+      {name: 'slash'},
+    ],
+  },
+  {
+    name: 'map',
+    description:
+      'makes it possible to look up the value associated with a key in a map, and much more',
+    functions: [
+      {name: 'get'},
+      {name: 'set'},
+      {name: 'merge'},
+      {name: 'remove'},
+      {name: 'keys'},
+      {name: 'values'},
+      {name: 'has-key'},
+      {name: 'deep-merge'},
+      {name: 'deep-remove'},
+    ],
+  },
+  {
+    name: 'math',
+    description: 'provides functions that operate on numbers',
+    variables: [
+      {name: '$e', description: undefined},
+      {name: '$pi'},
+      {name: '$epsilon'},
+      {name: '$max-safe-integer'},
+      {name: '$min-safe-integer'},
+      {name: '$max-number'},
+      {name: '$min-number'},
+    ],
+    functions: [
+      {name: 'ceil', description: undefined},
+      {name: 'clamp'},
+      {name: 'floor'},
+      {name: 'max'},
+      {name: 'min'},
+      {name: 'round'},
+      {name: 'abs'},
+      {name: 'hypot'},
+      {name: 'log'},
+      {name: 'pow'},
+      {name: 'sqrt'},
+      {name: 'acos'},
+      {name: 'asin'},
+      {name: 'atan'},
+      {name: 'atan2'},
+      {name: 'cos'},
+      {name: 'sin'},
+      {name: 'tan'},
+      {name: 'compatible'},
+      {name: 'is-unitless'},
+      {name: 'unit'},
+      {name: 'div'},
+      {name: 'percentage'},
+      {name: 'random'},
+    ],
+  },
+  {
+    name: 'meta',
+    description: 'exposes the details of Sass’s inner workings',
+    functions: [
+      {name: 'apply'},
+      {name: 'load-css'},
+      {name: 'accepts-content'},
+      {name: 'calc-args'},
```

Comment:
Missing `calc-name`?

```suggestion
      {name: 'calc-args'},
      {name: 'calc-name'},
```